calblueprint/siren-web
# Andrei/submit buttons

# ✨ New in this PR ✨

## One liner: what did you do? 🚨 
Connect intake forms to firebase

## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [x] Retrieval of questions from firebase.
 - [x] Send questions to firebase
 - [x] Mult-lingual functionality

## How can the reviewer test your code? 👩‍💻
_Run `npm run dev` and check that questions on intake form match firebase.   Update questions, create new ones, or mess with the ordering.  Click publish to send to firebase.  WARNING: After clicking publish, wait a few seconds before refreshing page to reflect changes, can corrupt the fields of the questions collection if refreshed too soon.  In case of FATAL ERROR when updating firebase, use this [link](https://console.cloud.google.com/firestore/import-export?project=siren-6099f) to import the original DacaRenewal questions.  Just get the meta data from sirenbackup/standardardizedDACAquestions and you should be good_ 

## Any bugs you encountered or still having trouble with? 🐛
_Issues with load-times when running getQuestion queries,  fixed by not using useStates with useReducer.  Unsure what parameter "examples" is supposed to fill in the question components.  "Required" toggle is universal, flipping it for one flips all of them.  Can be fixed but there are styling issues with this._

_Issues with sending information to firebase.  Cannot send maps directly, so fields dependent on that will at first be sent as JSON  strings (redundant) in the set() before being overwritten manually in the .then (can be seen in setQuestion query) to become the correct maps once again.  Risky if someone refreshes the page before the .then is complete, which leaves us with JSON strings instead of maps in the firebase, loading will be an issue if this happens._

## Resources 📔
_link online resources and related PRs.  
[DACArenewal backup](https://console.cloud.google.com/firestore/import-export?project=siren-6099f)_ 


🧜‍♀️ cc: @shannonbonet @phoebeli23
